app/models/service_template_ansible_playbook.rb

```diff
@@ -117,4 +109,25 @@ def self.validate_config_info(info)
     info
   end
   private_class_method :validate_config_info
+
+  def job_template(action)
+    resource_actions.find_by!(:action => action.to_s.capitalize).configuration_template
+  end
+
+  def update_catalog_item(options, auth_user = nil)
+    config_info = validate_update_config_info(options)
+    name = options[:name]
+    description = options[:description]
+    [:provision, :retirement, :reconfigure].each do |action|
+      next unless config_info[action]
+      info = config_info[action]
+      self.class.send(:create_new_dialog, info[:new_dialog_name], job_template(action)) if info[:new_dialog_name]
+
+      next unless info.keys.include?(:playbook_id)
+      tower, params = self.class.send(:build_parameter_list, name, description, info)
+      params[:manager_ref] = job_template(action).manager_ref
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript.update_in_provider_queue(tower.id, params, auth_user)
```

Comment:
We need to handle the case for example initially there was no playbook assigned to retirement, but through the update now a new playbook is given. We will also need to update the job template in the corresponding resource action,